src/client/components/date-range-picker/date-range-picker.tsx

```diff
@@ -154,13 +155,7 @@ export class DateRangePicker extends React.Component<DateRangePickerProps, DateR
 
   renderCalendar(startDate: Date): JSX.Element[] {
     const { timezone } = this.props;
-    const weeks: Date[][] = monthToWeeks(startDate, timezone, getLocale());
-    const firstWeek = weeks[0];
-    const lastWeek = weeks[weeks.length - 1];
-    const countPrepend = 7 - firstWeek.length;
-    const countAppend = 7 - lastWeek.length;
-    weeks[0] = prependDays(timezone, firstWeek, countPrepend);
-    weeks[weeks.length - 1] = appendDays(timezone, lastWeek, countAppend);
+    const weeks: Date[][] = calendarDays(startDate, timezone, getLocale());
```

Comment:
I really like this change, logic not in the react component but in separate functions.